label_studio/projects/models.py

```diff
@@ -1200,6 +1200,15 @@ def update_created_annotations_and_labels(self, annotations):
         self.save(update_fields=['created_annotations', 'created_labels'])
 
     def remove_created_annotations_and_labels(self, annotations):
+        # we are going to remove all drafts, make full refresh
+        if self.project.annotations.count() == annotations.count():
```

Comment:
Why do these need to be equal?

Reply:
this way we can know that all existing annotations in the project are going to be removed and it's a special case where we can totally reset created_labels = {}